torchtext/vocab/vocab_factory.py

```diff
@@ -95,5 +103,5 @@ def build_vocab_from_iterator(iterator: Iterable, min_freq: int = 1, specials: O
     ordered_dict = OrderedDict(sorted_by_freq_tuples)
```

Comment:
SInce we already using `Counter`, would it be more computationally efficient if we used the built-in method `counter.most_common(max_words)` before passing to the `vocab()` function?

Reply:
I was actually thinking the same thing as I made this PR. I think the `OrderedDict` is a vestige of pre-python 3.7 dictionary ordering. 

More to the point, the question is ultimately whether `vocab` should handle `max_tokens` or the `build_from_*`. Since `vocab` also handles the `min_freq` constraint, it made sense to push `max_tokens` there as well. Whether that's ideal is another question.

Reply:
Yeah totally makes sense. If `vocab` would accept a `max_tokens` then it would make sense to not duplicate the truncation in `build_vocab-from_iterator`.

Reply:
> More to the point, the question is ultimately whether `vocab` should handle `max_tokens` or the `build_from_*`. Since `vocab` also handles the `min_freq` constraint, it made sense to push `max_tokens` there as well. Whether that's ideal is another question.

some insights here https://github.com/pytorch/text/pull/1525#discussion_r786256295.